addons/stock/models/stock_scrap.py

```diff
@@ -72,16 +72,9 @@ def onchange_product_id(self):
             if self.picking_id:
                 for move_line in self.picking_id.move_line_ids:
                     if move_line.product_id == self.product_id:
-                        self.location_id = move_line.location_id
+                        self.location_id = move_line.location_id if move_line.state != 'done' else move_line.location_dest_id
```

Comment:
already done here i think https://github.com/odoo/odoo/commit/72c36aaf9004392773845406b2dc248017bb2e0c#diff-2e87b22b4f14124b45d586be67b759d4

Reply:
I am still having an issue. https://drive.google.com/file/d/1lgsZ-WU50Wf5eCmaRu64MRbBkzTGWDWM/view?usp=sharing

Reply:
ok my bad, i only checked the initial value of the wizard, not the one after selecting a product